src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/CounterAggregator.cs

```diff
@@ -16,21 +24,56 @@ public CounterAggregator(bool isMonotonic)
 
         public override void Update(double value)
         {
-            lock (this)
+            // Get the deltas array.
+            PaddedDouble[] deltas = _deltas;
+
+            // Get the delta best associated with the current thread, preferring to use core ID rather than
+            // thread ID to reduce contention.
+            ref PaddedDouble delta = ref deltas[
+#if NETCOREAPP2_1_OR_GREATER
+                Thread.GetCurrentProcessorId()
+#else
+                Environment.CurrentManagedThreadId
+#endif
+                % deltas.Length];
```

Comment:
Should we use `% Environment.ProcessorCount` here instead? Thus it will be converted to a jit-time constant and the whole mod operation will be cheaper